python/cudf/cudf/core/reshape.py

```diff
@@ -61,6 +62,7 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     # when axis is 1 (column) we can concat with Series and Dataframes
     if axis == 1:
         assert typs.issubset(allowed_typs)
+        objs = list(_align_indices(objs[0], objs[1]))
```

Comment:
`objs` can totally be longer than 2 items. We need to do a pairwise alignment here by looping through objs which can be done in the same loop as below on line 69

Reply:
I suspected, but I was thinking possibly we only needed to align the first set as the tests didn't complain. I'll add tests for larger concatenations too.

Reply:
Yea likely just a lack of test coverage, thanks